RecoHI/HiEvtPlaneAlgos/src/HiEvtPlaneFlatProducer.cc

```diff
@@ -167,62 +178,63 @@ void HiEvtPlaneFlatProducer::produce(edm::Event& iEvent, const edm::EventSetup&
       }
     }
   }
-
+  //
+  //Get flattening parameter file.
+  //
   if (hirpWatcher.check(iSetup)) {
     edm::ESHandle<RPFlatParams> flatparmsDB_;
     iSetup.get<HeavyIonRPRcd>().get(flatparmsDB_);
     LoadEPDB db(flatparmsDB_, flat);
-    if (!db.IsSuccess())
-      return;
-  }
+  }  //rp record change
+
   //
   //Get Centrality
   //
-
-  int bin = iEvent.get(centralityBinToken) / CentBinCompression_;
-
-  if (Noffmin_ >= 0) {
-    int nOff = iEvent.get(centralityToken).Ntracks();
-    if ((nOff < Noffmin_) or (nOff >= Noffmax_)) {
-      return;
-    }
-  }
+  int bin = 0;
+  int cbin = 0;
+  cbin = iEvent.get(centralityBinToken_);
+  bin = cbin / CentBinCompression_;
   //
   //Get Vertex
   //
-  int vs_sell;  // vertex collection size
-  float vzr_sell;
-  auto const& vertices3 = iEvent.get(vertexToken);
-  vs_sell = vertices3.size();
-  if (vs_sell > 0) {
-    vzr_sell = vertices3.begin()->z();
-  } else
-    vzr_sell = -999.9;
+
+  //best vertex
+  double bestvz = -999.9;
+  const reco::Vertex& vtx = iEvent.get(vertexToken_)[0];
+  bestvz = vtx.z();
+  //  Produce the EP regardless of vz position
+  //  if (bestvz < minvtx_ || bestvz > maxvtx_)
+  //    return;
 
   //
   //Get Event Planes
   //
 
+  edm::Handle<reco::EvtPlaneCollection> evtPlanes_;
+  iEvent.getByToken(inputPlanesToken_, evtPlanes_);
```

Comment:
```suggestion
  auto const& evtPlanes = iEvent.get(inputPlanesToken_);
```
- trailing underscore is reserved for data members
- shorter access syntax can be used (drop `*` in `*evtPlanes` below)